examples/CMakeLists.txt

```diff
@@ -6,6 +6,17 @@ ELSE()
 	TARGET_LINK_LIBRARIES(cgit2 git2 pthread)
 ENDIF()
 
+# Use -Wno-declaration-after-statement -Wunused-but-set-variable on global.c
+FOREACH(warn declaration-after-statement unused-but-set-variable)
+	SET(no_warn "-Wno-${warn}")
+	STRING(TOUPPER ${warn} _UPCASE)
+	SET(_var "IS_WNO_${_UPCASE}_SUPPORTED")
+	CHECK_C_COMPILER_FLAG(${no_warn} ${_var})
+	IF(${_var})
+		SET_PROPERTY(SOURCE general.c APPEND_STRING PROPERTY COMPILE_FLAGS " ${no_warn}")
+	ENDIF()
+ENDFOREACH()
+
```

Comment:
Declarations don't have to be at the top of a function, they have to be at the top of a block. You can create blocks anywhere. But these are code snippets that aren't meant to be run anyhow, they're just there for a bit of demonstration.